Do not emit 'up' on 'drop'

tibotiber opened the pull request:
Closes #66 

@williamngan here you go, I fixed it not at the `UIDragger` level but at the `Space` level since space events "suffered the same fate" 😄. 

Hope it looks good to you. One "major" change is some click event are not emitted anymore. Is there any other place I might not know about that relies on them?

williamngan commented:
Hi @tibotiber , thanks for you help on this! I tested this in various case. It seems to work well except in this case -- 

Try the following in the [ui.track demo](https://ptsjs.org/demo/?name=ui.track)
- Click on the red dot (without moving)
- The dragging state is active, so when you move the pointer (without dragging), it acts as if dragging
- Click again and the drop is ignored. Which means you cannot release the dragged item.

Any suggestions on how to fix this?

Unrelated, but I also found and fixed another bug. In [`unhold()`](https://github.com/williamngan/pts/commit/fed4d1e3d189aa00c850f06dabf2c82034e29d7b#diff-58099af40771a03d1a2eca783196326eL201), the javascript `splice` function actually would return the deleted items (javascript still caught me off guard sometimes! 😅), so I was wrong to assign it back to the array. It's now fixed in https://github.com/williamngan/pts/commit/fed4d1e3d189aa00c850f06dabf2c82034e29d7b so please pull the latest from master first.

Thank you!

tibotiber commented:
Hi @williamngan, good catch! I've fixed it by starting the drag logic on `drag` instead of `down`, mirroring the stop on `drop` instead of `up`. Seems to work from my tests but will see if you catch anything else.

I've rebased on top of master as well, thanks for the heads up.

williamngan commented:
Thanks @tibotiber for the fixes. This should be good to go!
